test/chess/store/game_test.exs

```diff
@@ -5,27 +5,34 @@ defmodule Chess.GameTest do
     alias Chess.Store.Game
 
     test "game is valid with a board and user" do
-      attrs = %{board: %{}, user_id: 1}
+      attrs = %{board: %{}, user_id: 1, opponent_id: 2}
```

Comment:
this reveals that you don't have [foreign_key_constraints](https://hexdocs.pm/ecto/Ecto.Changeset.html#foreign_key_constraint/3) enforced in your changeset. You might consider that. If you do end up enforcing keys, you'll have to re-write your tests, and ex_machina can help with [params_with_assocs](https://hexdocs.pm/ex_machina/ExMachina.Ecto.html#c:params_with_assocs/2) or [string_params_with_assocs](https://hexdocs.pm/ex_machina/ExMachina.Ecto.html#c:string_params_with_assocs/2)

Reply:
I've added the foreign_key_constraint's to the changesets, but the tests are still passing.

```elixir
  def create_changeset(struct, params \\ %{}) do
    struct
    |> cast(params, required_attrs())
    |> foreign_key_constraint(:user_id)
    |> foreign_key_constraint(:opponent_id)
    |> put_change(:board, Board.default)
    |> validate_required(required_attrs())
  end

  defp required_attrs, do: ~w[board user_id opponent_id]a
```

Reply:
Whoops! My mistake. Good improvement anyway though.